rust/src/krb/krb5.rs

```diff
@@ -158,7 +187,10 @@ impl KRB5State {
                         let res = krb5_parser::parse_krb_error(i);
                         if let Ok((_,error)) = res {
                             let mut tx = self.new_tx();
-                            tx.msg_type = MessageType(self.req_id as u32);
+                            if self.req_id > 0 {
```

Comment:
Why only if `self.req_id > 0` ?

Reply:
in the code self.req_id is only set different to zero for requests, so you only get the request type when the previous one was a request, in other case, you keep the default None Option.

Reply:
ok